packages/nimble-components/src/rich-text/models/tests/markdown-parser.spec.ts

```diff
@@ -923,4 +931,324 @@ describe('Markdown parser', () => {
+                'P',
+                `${anchorTag}`.toUpperCase()
+            ]);
+            expect(lastChildElementHasAttribute('href', doc)).toBeFalse();
+            expect(getLeafContentsFromElement(doc)).toEqual(['issue:1']);
+        });
+
+        it('should get anchor element with no href when display name is missing', async () => {
+            ({ element, connect, disconnect } = await setup(
+                [{ key: 'user:1' }, { key: 'user:2' }],
+                '^user:.*'
+            ));
+            await connect();
+            const doc = RichTextMarkdownParser.parseMarkdownToDOM('<user:1>', [
+                new MarkdownParserMentionConfiguration(element.mentionInternals)
+            ]);
+
+            expect(getTagsFromElement(doc)).toEqual([
+                'P',
+                `${anchorTag}`.toUpperCase()
+            ]);
+            expect(lastChildElementHasAttribute('href', doc)).toBeFalse();
+            expect(getLeafContentsFromElement(doc)).toEqual(['user:1']);
+        });
+
+        it('should get anchor element with no href when the pattern does not match the autolink in markdown', async () => {
+            ({ element, connect, disconnect } = await setup(
+                [
+                    { key: 'user:1', displayName: 'username1' },
+                    { key: 'user:2', displayName: 'username1' }
+                ],
+                'abc'
+            ));
+            await connect();
+            const doc = RichTextMarkdownParser.parseMarkdownToDOM('<user:1>', [
+                new MarkdownParserMentionConfiguration(element.mentionInternals)
+            ]);
+
+            expect(getTagsFromElement(doc)).toEqual([
+                'P',
+                `${anchorTag}`.toUpperCase()
+            ]);
+            expect(lastChildElementHasAttribute('href', doc)).toBeFalse();
+            expect(getLeafContentsFromElement(doc)).toEqual(['user:1']);
+        });
+
+        it('should get anchor element with no href when keys do not match the pattern', async () => {
+            ({ element, connect, disconnect } = await setup(
+                [
+                    { key: 'abc', displayName: 'username1' },
+                    { key: 'abc', displayName: 'username1' }
+                ],
+                '^user:.*'
+            ));
+            await connect();
+            const doc = RichTextMarkdownParser.parseMarkdownToDOM('<user:1>', [
+                new MarkdownParserMentionConfiguration(element.mentionInternals)
+            ]);
+
+            expect(getTagsFromElement(doc)).toEqual([
+                'P',
+                `${anchorTag}`.toUpperCase()
+            ]);
+            expect(lastChildElementHasAttribute('href', doc)).toBeFalse();
+            expect(getLeafContentsFromElement(doc)).toEqual(['user:1']);
+        });
+
+        it('should get anchor element with href when autolink markdown format is HTTP but does not match with the pattern', async () => {
```

Comment:
I think we're missing a case where the mention `pattern` matches certain HTTP or HTTPS URLs but not others. For example, if the pattern is `^http://user/(.*)` and the contents are `<http://ni.com>` then it should render as a link, not a mention.

Reply:
Added the suggested test case.